reaktive/src/commonMain/kotlin/com/badoo/reaktive/single/RepeatUntil.kt

```diff
@@ -0,0 +1,44 @@
+package com.badoo.reaktive.single
+
+import com.badoo.reaktive.base.ErrorCallback
+import com.badoo.reaktive.base.tryCatch
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.observable.Observable
+import com.badoo.reaktive.observable.observable
+import com.badoo.reaktive.utils.serializer.serializer
+
+fun <T> Single<T>.repeatUntil(predicate: (T) -> Boolean): Observable<T> =
+    observable { emitter ->
+        val observer =
+            object : SingleObserver<T>, ErrorCallback by emitter {
+                // Prevents recursive subscriptions
+                private val serializer =
+                    serializer<Unit> {
+                        subscribe(this)
+                        true
+                    }
+
+                override fun onSubscribe(disposable: Disposable) {
+                    emitter.setDisposable(disposable)
+                }
+
+                override fun onSuccess(value: T) {
+                    emitter.onNext(value)
+
+                    emitter.tryCatch(
+                        block = { predicate(value) },
+                        onSuccess = { shouldRepeat ->
```

Comment:
The name of `shouldRepeat` should be opposite, e.g. `shouldComplete`. Or just remove and use `it` as with `Observable` case.

Reply:
sorry my bad

Reply:
Updated PR